test/lightning/workflows_test.exs

```diff
@@ -207,4 +209,108 @@ defmodule Lightning.WorkflowsTest do
       assert Jobs.get_job!(job_2.id).enabled == false
     end
   end
+
+  describe "Project digest" do
+    test "Gets project digest data" do
+      user = AccountsFixtures.user_fixture()
+
+      project =
+        ProjectsFixtures.project_fixture(
+          project_users: [%{user_id: user.id, digest: :daily}]
+        )
+
+      workflow = WorkflowsFixtures.workflow_fixture(project_id: project.id)
+
+      job =
+        JobsFixtures.job_fixture(
+          project_id: project.id,
+          workflow_id: workflow.id
+        )
+
+      workorder_1 =
+        InvocationFixtures.work_order_fixture(workflow_id: workflow.id)
+
+      workorder_2 =
+        InvocationFixtures.work_order_fixture(workflow_id: workflow.id)
+
+      reason = InvocationFixtures.reason_fixture(trigger_id: job.trigger.id)
+
+      create_run(workorder_1, reason, %{
+        project_id: project.id,
+        job_id: job.id,
+        input_dataclip_id: reason.dataclip_id,
+        exit_code: 0,
+        finished_at: Timex.now()
+      })
+
+      create_run(workorder_2, reason, %{
+        project_id: project.id,
+        job_id: job.id,
+        input_dataclip_id: reason.dataclip_id,
+        exit_code: 0,
+        finished_at: Timex.now()
+      })
+
+      assert Workflows.get_digest_data(
+               workflow |> Repo.preload(:work_orders),
+               :daily
+             ) == %{
+               failed_workorders: 0,
+               rerun_workorders: 0,
+               successful_workorders: 2,
+               workflow_name: workflow.name
+             }
+
+      workorder_1 =
+        InvocationFixtures.work_order_fixture(workflow_id: workflow.id)
+
+      workorder_2 =
+        InvocationFixtures.work_order_fixture(workflow_id: workflow.id)
+
+      workorder_3 =
+        InvocationFixtures.work_order_fixture(workflow_id: workflow.id)
+
+      create_run(workorder_1, reason, %{
+        project_id: project.id,
+        job_id: job.id,
+        input_dataclip_id: reason.dataclip_id,
+        exit_code: 0,
+        finished_at: Timex.now()
+      })
+
+      create_run(workorder_2, reason, %{
+        project_id: project.id,
+        job_id: job.id,
+        input_dataclip_id: reason.dataclip_id,
+        exit_code: 1,
+        finished_at: Timex.now()
+      })
+
+      create_run(workorder_3, reason, %{
+        project_id: project.id,
+        job_id: job.id,
+        input_dataclip_id: reason.dataclip_id,
+        exit_code: 1,
+        finished_at: Timex.now()
+      })
+
+      assert Workflows.get_digest_data(
+               workflow |> Repo.preload(:work_orders),
+               :daily
+             ) == %{
+               failed_workorders: 2,
+               rerun_workorders: 0,
+               successful_workorders: 3,
+               workflow_name: workflow.name
+             }
```

Comment:
All of these scenarios need to have data to _prove_ they aren't including other runs... Add some data you expect to not be found.